File: desktop/src/types/kanban.ts
import { ColumnId } from '../config/kanbanColumns';

export interface GhRepo {
  nameWithOwner: string;
  name: string;
  owner: { login: string };
  description: string | null;
  url: string;
  isPrivate: boolean;
  isArchived: boolean;
  pushedAt: string | null;
}

export interface KanbanCard {
  nameWithOwner: string;
  column: ColumnId;
  createdAt: number;
  updatedAt: number;
}

export interface KanbanState {
  version: number;
  cards: Record<string, KanbanCard>;
}

export type SyncStatus = 'disabled' | 'synced' | 'offline' | 'expired';

export interface KanbanRefresh {
  repos: GhRepo[];
  state: KanbanState;
  syncStatus: SyncStatus;
}

export type GhAuthStatus =
  | { status: 'ok'; user: string }
  | { status: 'notInstalled' }
  | { status: 'notAuthenticated' }
  | { status: 'error'; message: string };

export interface KanbanCardView {
  card: KanbanCard;
  repo: GhRepo;
}

File: desktop/src-tauri/src/domain/kanban.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct KanbanCard {
    pub name_with_owner: String,
    pub column: String,
    pub created_at: i64,
    pub updated_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KanbanState {
    pub version: u32,
    pub cards: HashMap<String, KanbanCard>,
}

impl Default for KanbanState {
    fn default() -> Self {
        Self {
            version: 2,
            cards: HashMap::new(),
        }
    }
}
